File: snippets/go/snippet-02.go
package main

import (
  "context"
  "fmt"
  "log"
  "os"

  apiclientgo "github.com/gleanwork/api-client-go"
  "github.com/gleanwork/api-client-go/models/components"
)

func main() {
  ctx := context.Background()

  s := apiclientgo.New(
    apiclientgo.WithSecurity(os.Getenv("GLEAN_API_TOKEN")),
    apiclientgo.WithInstance(os.Getenv("GLEAN_INSTANCE")),
  )

  res, err := s.Client.Chat.Create(ctx, components.ChatRequest{
    Messages: []components.ChatMessage{
      {
        Fragments: []components.ChatMessageFragment{
          {Text: apiclientgo.Pointer("What are our company values?")},
        },
      },
    },
  }, nil, nil)
  if err != nil {
    log.Fatal(err)
  }

  if res.ChatResponse != nil {
    for _, msg := range res.ChatResponse.Messages {
      for _, frag := range msg.Fragments {
        if frag.Text != nil {
          fmt.Print(*frag.Text)
        }
      }
    }
    fmt.Println()
  }
}

File: snippets/java/snippet-02.java
import com.glean.api_client.glean_api_client.Glean;
import com.glean.api_client.glean_api_client.models.components.*;
import java.util.List;

public class GleanExample {
  public static void main(String[] args) throws Exception {
    Glean client = Glean.builder()
        .apiToken(System.getenv("GLEAN_API_TOKEN"))
        .serverURL(System.getenv("GLEAN_SERVER_URL"))
        .build();

    var response = client.client().chat().create()
        .chatRequest(ChatRequest.builder()
            .messages(List.of(
                ChatMessage.builder()
                    .fragments(List.of(
                        ChatMessageFragment.builder()
                            .text("What are our company values?")
                            .build()))
                    .build()))
            .build())
        .call();

    if (response.chatResponse().isPresent()) {
      System.out.println(response.chatResponse().get());
    }
  }
}
